### PR TITLE
[DUOS-2480][risk=no]Fix DataUseMatchPairResultV3 type

### PR DESCRIPTION
**Pull request by kaitlynoflaherty**
## Addresses
https://broadworkbench.atlassian.net/jira/software/c/projects/DUOS/boards/123?modal=detail&selectedIssue=DUOS-2480

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by fboulnois — APPROVED** [review]
I wonder if we could add a unit test for this.

**Review by connorlbark — APPROVED** [review]

File: src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java [hunk 1]
```diff
@@ -114,18 +115,75 @@ public void testOKV3() {
...
     DataUseV3 dataset = new DataUseBuilderV3().setGeneralUse(true).build();
     DataUseMatchPairV3 pair = new DataUseMatchPairV3(purpose, dataset);
-    try( Response response = resource.matchDataUseV3(pair)){
+    try (Response response = resource.matchDataUseV3(pair)) {
       assertEquals(200, response.getStatus());
     }
   }
 
+  @Test
+  public void testOKV3Response_approve() {
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java]:
```suggestion
  public void testOKV3ResponseApprove() {
```

File: src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java [hunk 2]
```diff
@@ -114,18 +115,75 @@ public void testOKV3() {
...
+      String stringEntity = response.getEntity().toString();
+      MatchV3ResponseEntity entity = new Gson().fromJson(stringEntity, MatchV3ResponseEntity.class);
+      assertEquals(MatchResultType.APPROVE, entity.getResult());
+      assertEquals(pair, entity.getMatchPair());
+      assertEquals(Collections.emptyList(), entity.getFailureReasons());
+    }
+  }
+
+  @Test
+  public void testOKV3Response_deny() {
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java]:
```suggestion
  public void testOKV3ResponseDeny() {
```

File: src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java [hunk 3]
```diff
@@ -114,18 +115,75 @@ public void testOKV3() {
...
+      String stringEntity = response.getEntity().toString();
+      MatchV3ResponseEntity entity = new Gson().fromJson(stringEntity, MatchV3ResponseEntity.class);
+      assertEquals(result_deny, entity.getResult());
+      assertEquals(pair, entity.getMatchPair());
+      assertEquals(Collections.emptyList(), entity.getFailureReasons());
+    }
+  }
+
+  @Test
+  public void testOKV3Response_abstain() {
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java]:
```suggestion
  public void testOKV3ResponseAbstain() {
```

File: src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java [hunk 4]
```diff
@@ -114,18 +115,75 @@ public void testOKV3() {
...
+      assertEquals(MatchResultType.APPROVE, entity.getResult());
+      assertEquals(pair, entity.getMatchPair());
+      assertEquals(Collections.emptyList(), entity.getFailureReasons());
+    }
+  }
+
+  @Test
+  public void testOKV3Response_deny() {
+    initResource();
+    MatchResultType result_deny = MatchResultType.DENY;
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java]:
```suggestion
    MatchResultType deny = MatchResultType.DENY;
```

File: src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java [hunk 5]
```diff
@@ -114,18 +115,75 @@ public void testOKV3() {
...
+    when(
+        dataUseMatcherV3.matchPurposeAndDatasetV3(any(DataUseV3.class),
+            any(DataUseV3.class))).thenReturn(
+        new MatchResult(result_deny, Collections.emptyList())
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java]:
```suggestion
        new MatchResult(deny, Collections.emptyList())
```

File: src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java [hunk 6]
```diff
@@ -114,18 +115,75 @@ public void testOKV3() {
...
+    );
+    DataUseV3 purpose = new DataUseBuilderV3().setSecondaryOther("true").build();
+    DataUseV3 dataset = new DataUseBuilderV3().setPopulationOriginsAncestry(true).build();
+    DataUseMatchPairV3 pair = new DataUseMatchPairV3(purpose, dataset);
+    try (Response response = resource.matchDataUseV3(pair)) {
+      String stringEntity = response.getEntity().toString();
+      MatchV3ResponseEntity entity = new Gson().fromJson(stringEntity, MatchV3ResponseEntity.class);
+      assertEquals(result_deny, entity.getResult());
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java]:
```suggestion
      assertEquals(deny, entity.getResult());
```

File: src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java [hunk 7]
```diff
@@ -114,18 +115,75 @@ public void testOKV3() {
...
+      assertEquals(pair, entity.getMatchPair());
+      assertEquals(Collections.emptyList(), entity.getFailureReasons());
+    }
+  }
+
+  @Test
+  public void testOKV3Response_abstain() {
+    initResource();
+    MatchResultType result_abstain = MatchResultType.ABSTAIN;
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java]:
```suggestion
    MatchResultType abstain = MatchResultType.ABSTAIN;
```

File: src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java [hunk 8]
```diff
@@ -114,18 +115,75 @@ public void testOKV3() {
...
+    when(
+        dataUseMatcherV3.matchPurposeAndDatasetV3(any(DataUseV3.class),
+            any(DataUseV3.class))).thenReturn(
+        new MatchResult(result_abstain, Collections.emptyList())
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java]:
```suggestion
        new MatchResult(abstain, Collections.emptyList())
```

File: src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java [hunk 9]
```diff
@@ -114,18 +115,75 @@ public void testOKV3() {
...
+    );
+    DataUseV3 purpose = new DataUseBuilderV3().setSecondaryOther("true").build();
+    DataUseV3 dataset = new DataUseBuilderV3().setPopulationOriginsAncestry(true).build();
+    DataUseMatchPairV3 pair = new DataUseMatchPairV3(purpose, dataset);
+    try (Response response = resource.matchDataUseV3(pair)) {
+      String stringEntity = response.getEntity().toString();
+      MatchV3ResponseEntity entity = new Gson().fromJson(stringEntity, MatchV3ResponseEntity.class);
+      assertEquals(result_abstain, entity.getResult());
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResourceTest.java]:
```suggestion
      assertEquals(abstain, entity.getResult());
```

**Review by fboulnois — APPROVED** [review]
Minor suggestions for formatting:

**Pull request merged**